### ISSUE TITLE
Does react-tracking record impression?

### ISSUE THREAD

**Issue by khieu** [NYTimes/react-tracking#64]
Hello,

I found react-tracking while looking for a solution to track clicks and impressions on a react web application. After some reading it looks like react-tracking handles click events. I want to ask if this library also help record impression events (a part/component of a page shows up on the screen) ? If the library does support that where can I find docs and example ?

Thank you so much for your help, really appreciate it !

**Comment by tizmagik** [comment]
Yes, it's possible to use react-tracking for that, but it's left up to the app to fire off the tracking call since what defines an "impression" varies by app.

For example here's how we're doing impression tracking in one case. This makes use of [react-visibility-sensor](https://www.npmjs.com/package/react-visibility-sensor).

```js
 onChange = isVisible => {
    if (isVisible) {
      this.props.tracking.trackEvent({
        event: 'impression',
      });

      this.setState({
        isVisible: true,
      });
    }
  };

  render() {
    return (
        <div>
          <VisibilitySensor
            scrollCheck
            scrollThrottle={100}
            intervalDelay={8000}
            onChange={this.onChange}
          />
          {this.state.isVisible && <FooComponent />}
        </div>
    );
  }
```

Hope that helps @khieu !
